.github/workflows/ci.yml

```
@@ -1,172 +1,91 @@
+        channel: [beta, nightly]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     env:
-      DISPLAY: ":99"
       ATOM_GITHUB_BABEL_ENV: coverage
-      ATOM_CHANNEL: nightly
-      ATOM: /tmp/atom/usr/bin/atom-nightly
-      APM: /tmp/atom/usr/share/atom-nightly/resources/app/apm/bin/apm
       MOCHA_TIMEOUT: 60000
       UNTIL_TIMEOUT: 30000
     steps:
     - uses: actions/checkout@v1
     - name: install Atom
-      shell: bash
-      run: |
-        curl -s -L https://atom.io/download/deb?channel=${ATOM_CHANNEL} \
-          -H 'Accept: application/octet-stream' \
-          -o /tmp/atom-amd64.deb
-        /sbin/start-stop-daemon --start --quiet \
-          --pidfile /tmp/custom_xvfb_99.pid \
-          --make-pidfile --background \
-          --exec /usr/bin/Xvfb \
-          -- :99 -ac -screen 0 1280x1024x16
-        sudo apt-get update
-        sudo apt-get install -yyq libgconf-2-4 build-essential git libsecret-1-dev gvfs-bin
-        dpkg-deb -x /tmp/atom-amd64.deb /tmp/atom
-    - name: install dependencies
-      shell: bash
-      run: ${APM} ci
-    - name: configure git
-      shell: bash
-      run: |
-        git config --global user.name Hubot
-        git config --global user.email hubot@github.com
-    - name: run tests
-      shell: bash
-      run: ${ATOM} --test test/
-    - name: report code coverage
-      shell: bash
-      env:
-        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-        SYSTEM_PULLREQUEST_PULLREQUESTNUMBER: ${{ github.event.number }}
-        SYSTEM_PULLREQUEST_SOURCEBRANCH: ${{ github.head_ref }}
-        BUILD_SOURCEBRANCH: ${{ github.event.ref }}
-      run: |
-        npm run report:coverage
-        bash <(curl -s https://codecov.io/bash) \
-          -n "Linux" \
-          -P "${SYSTEM_PULLREQUEST_PULLREQUESTNUMBER:-}" \
-          -B "${SYSTEM_PULLREQUEST_SOURCEBRANCH:-${BUILD_SOURCEBRANCH}}"
-      if: success() || failure()
+      uses: UziTech/action-setup-atom@v1
+      with:
+        channel: ${{ matrix.channel }}
 
-  macos-tests:
-    name: macos tests
-    runs-on: macos-10.14
-    env:
-      ATOM_GITHUB_BABEL_ENV: coverage
-      ATOM_CHANNEL: nightly
-      ATOM: /tmp/atom/Atom\ Nightly.app/Contents/Resources/app/atom.sh
-      APM: /tmp/atom/Atom\ Nightly.app/Contents/Resources/app/apm/bin/apm
-      MOCHA_TIMEOUT: 60000
-      UNTIL_TIMEOUT: 30000
-    steps:
-    - uses: actions/checkout@v1
-    - name: install Atom
-      shell: bash
+    - name: install windows build tools
+      if: contains(matrix.os, 'windows')
       run: |
-        curl -s -L https://atom.io/download/mac?channel=${ATOM_CHANNEL} \
-          -H 'Accept: application/octet-stream' \
-          -o /tmp/atom.zip
-        sudo unzip -q /tmp/atom.zip -d /tmp/atom
+        choco install visualcpp-build-tools --version=14.0.25420.1 --ignore-dependencies -y --params "'/IncludeRequired'"
+        echo ::set-env name=VCTargetsPath::'C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\v140'
```

The old command is deprecated now. 

```suggestion
        echo "VCTargetsPath='C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\v140'" >> $GITHUB_ENV
```